homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_zone_control_zone.py

```diff
@@ -0,0 +1,252 @@
+class AtlanticPassAPCZoneControlZone(AtlanticPassAPCHeatingZone):
+    """Representation of Atlantic Pass APC Heating And Cooling Zone Control."""
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+
+        # There is less supported functions, because they depend on the ZoneControl.
+        if not self.is_using_derogated_temperature_fallback:
+            # Modes are not configurable, they will follow current HVAC Mode of Zone Control.
+            self._attr_hvac_modes = []
+
+            # Those are available and tested presets on Shogun.
+            self._attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+
+        # Those APC Heating and Cooling probes depends on the zone control device (main probe).
+        # Only the base device (#1) can be used to get/set some states.
+        # Like to retrieve and set the current operating mode (heating, cooling, drying, off).
+        self.zone_control_device = self.executor.linked_device(
+            TEMPERATURE_ZONECONTROL_DEVICE_INDEX
+        )
+
+    @property
+    def is_using_derogated_temperature_fallback(self) -> bool:
+        """Check if the device behave like the Pass APC Heating Zone."""
+
+        return self.executor.has_command(
+            OverkizCommand.SET_DEROGATED_TARGET_TEMPERATURE
+        )
+
+    @property
+    def zone_control_hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool, dry, off mode."""
+
+        if (
+            state := self.zone_control_device.states[
+                OverkizState.IO_PASS_APC_OPERATING_MODE
+            ]
+        ) is not None and (value := state.value_as_str) is not None:
+            return OVERKIZ_TO_HVAC_MODE[value]
+        return HVACMode.OFF
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool, dry, off mode."""
+
+        if self.is_using_derogated_temperature_fallback:
+            return super().hvac_mode
+
+        zone_control_hvac_mode = self.zone_control_hvac_mode
+
+        # Should be same, because either thermostat or this integration change both.
+        on_off_state = cast(
+            str,
+            self.executor.select_state(
+                OverkizState.CORE_COOLING_ON_OFF
+                if zone_control_hvac_mode == HVACMode.COOL
+                else OverkizState.CORE_HEATING_ON_OFF
+            ),
+        )
+
+        # Device is Stopped, it means the air flux is flowing but its venting door is closed.
+        if on_off_state == OverkizCommandParam.OFF:
+            hvac_mode = HVACMode.OFF
+        else:
+            hvac_mode = zone_control_hvac_mode
+
+        # It helps keep it consistent with the Zone Control, within the interface.
+        if self._attr_hvac_modes != [zone_control_hvac_mode, HVACMode.OFF]:
+            self._attr_hvac_modes = [zone_control_hvac_mode, HVACMode.OFF]
```

Comment:
Changing the hvac modes during the entity lifetime is weird.

Regardless, there shouldn't be side effects in entity state properties. If another attribute needs to change it should be done in a state update callback.

Reply:
@MartinHjelmare yes it's as weird as those devices can be...

The thing is, none of those subsidiary devices can control the current mode that is given by the parent ZoneControl device. There only choices are On or Stop.

Maybe the best choice is to always use HEAT_COOL + OFF for those subsidiary devices.
I made the #111830 PR, for now I only use HEAT_COOL for AUTO, but I can always use it instead. It's just not this relevant to change cooling temperature when ZoneControl mode is forced to Heating.

Do you have any advice or tip ?

Reply:
Side note: The auto mode should be used only if the temperature can't be controlled by the user.

Why not only have on/off as mode for the simple devices?